src/NUnitFramework/tests/Assertions/AssertMultipleTests.cs

```diff
@@ -25,101 +25,89 @@
-        [TestCase("TwoAsserts_SecondAssertFails", "ImaginaryPart")]
-        [TestCase("TwoAsserts_BothAssertsFail", "RealPart", "ImaginaryPart")]
-        [TestCase("NestedBlock_FirstAssertFails", "Expected: 5")]
-        [TestCase("NestedBlock_TwoAssertsFail", "Expected: 5", "ImaginaryPart")]
-        [TestCase("TwoNestedBlocks_FirstAssertFails", "Expected: 5")]
-        [TestCase("TwoNestedBlocks_TwoAssertsFail", "Expected: 5", "ImaginaryPart")]
-        [TestCase("MethodCallsFail", "Message from Assert.Fail")]
-        [TestCase("MethodCallsFailAfterTwoAssertsFail", "Expected: 5", "ImaginaryPart", "Message from Assert.Fail")]
+        [TestCase(nameof(AM.TwoAsserts_FirstAssertFails), 2, "RealPart")]
+        [TestCase(nameof(AM.TwoAsserts_SecondAssertFails), 2, "ImaginaryPart")]
+        [TestCase(nameof(AM.TwoAsserts_BothAssertsFail), 2, "RealPart", "ImaginaryPart")]
+        [TestCase(nameof(AM.NestedBlock_FirstAssertFails), 3, "Expected: 5")]
+        [TestCase(nameof(AM.NestedBlock_TwoAssertsFail), 3, "Expected: 5", "ImaginaryPart")]
+        [TestCase(nameof(AM.TwoNestedBlocks_FirstAssertFails), 3, "Expected: 5")]
+        [TestCase(nameof(AM.TwoNestedBlocks_TwoAssertsFail), 3, "Expected: 5", "ImaginaryPart")]
+        [TestCase(nameof(AM.MethodCallsFail), 0, "Message from Assert.Fail")]
+        [TestCase(nameof(AM.MethodCallsFailAfterTwoAssertsFail), 2, "Expected: 5", "ImaginaryPart", "Message from Assert.Fail")]
+        [TestCase(nameof(AM.TwoAssertsFailAfterWarning), 2, "WARNING", "Expected: 5", "ImaginaryPart")]
+        [TestCase(nameof(AM.WarningAfterTwoAssertsFail), 2, "Expected: 5", "ImaginaryPart", "WARNING")]
 #if ASYNC
-        [TestCase("TwoAsserts_BothAssertsFail_Async", "RealPart", "ImaginaryPart")]
-        [TestCase("TwoNestedBlocks_TwoAssertsFail_Async", "Expected: 5", "ImaginaryPart")]
+        [TestCase(nameof(AM.TwoAsserts_BothAssertsFail_Async), 2, "RealPart", "ImaginaryPart")]
+        [TestCase(nameof(AM.TwoNestedBlocks_TwoAssertsFail_Async), 3, "Expected: 5", "ImaginaryPart")]
 #endif
-        public void AssertMultipleFails(string methodName, params string[] assertionMessageRegex)
+        public void AssertMultipleFails(string methodName, int asserts, params string[] assertionMessages)
         {
-            CheckResult(methodName, ResultState.Failure, assertionMessageRegex);
+            CheckResult(methodName, ResultState.Failure, asserts, assertionMessages);
         }
 
-        [TestCase("ExceptionThrown")]
-        [TestCase("ExceptionThrownAfterTwoFailures", "Failure 1", "Failure 2", "Simulated Error")]
-        public void AssertMultipleErrorTests(string methodName, params string[] assertionMessageRegex)
+        [TestCase(nameof(AM.ThreeAssertWarns), 0, "WARNING1", "WARNING2", "WARNING3")]
+        [TestCase(nameof(AM.ThreeWarnIf_TwoFail), 3, "WARNING1", "WARNING3")]
+        [TestCase(nameof(AM.ThreeWarnUnless_TwoFail), 3, "WARNING1", "WARNING3")]
```

Comment:
A question about Warn (which I've not used much - if at all). So `Assert.Warn` does not count as an assertion, but `Warn.If` and `War.Unless` do?

Reply:
I just checked it, and so it is.

Reply:
Same is true of `Assert.Fail`, `Assert.Pass` and `Assert.Inconclusive`.  I actually expected `Assert.Warn` to be counted so I went through the same research myself. 😄